decoder.js

```diff
@@ -136,8 +138,18 @@ GIFDecoder.prototype._readPalette = function() {
     
   this._palette = buf.slice(0, len);
   buf.consume(len);
-  
-  this._state = this._state === GIF_GLOBAL_PALETTE ? GIF_BLOCK : GIF_LZW;
+
+  if (this._state === GIF_GLOBAL_PALETTE) {
+    if (this._outputIndexed) {
+      this.format.palette = this._palette.slice(0);
```

Comment:
Ya, maybe not. I guess they ensure immutability of the palette when it is exposed to the outside world. If for some reason streams down the line mutated the palette it could cause odd looking gifs for whichever stream that palette came from. Whether this is important enough to merit the perf hit, donno, guess that's your call